src/client/client.cpp

```diff
@@ -607,10 +608,10 @@ void Client::step(float dtime)
 				v3s16 ofs;
 
 				// See also mapblock_mesh.cpp for the code that creates the array of minimap blocks.
-				for (ofs.Z = 0; ofs.Z <= 1; ofs.Z++)
-				for (ofs.Y = 0; ofs.Y <= 1; ofs.Y++)
-				for (ofs.X = 0; ofs.X <= 1; ofs.X++) {
-					size_t i = ofs.Z * 4 + ofs.Y * 2 + ofs.X;
+				for (ofs.Z = 0; ofs.Z < m_mesh_chunk; ofs.Z++)
+				for (ofs.Y = 0; ofs.Y < m_mesh_chunk; ofs.Y++)
+				for (ofs.X = 0; ofs.X < m_mesh_chunk; ofs.X++) {
+					size_t i = ofs.Z * m_mesh_chunk * m_mesh_chunk + ofs.Y * m_mesh_chunk + ofs.X;
```

Comment:
Would it be beneficial to wrap m_mesh_chunk into a small class with all the calculation methods? This line of code could also be added to avoid duplication across different files.

Reply:
Hmm... This part is quick specific to just minimap. Lemme think on this.

Reply:
I don't think it would make it more readable.

You'd have `m_mesh_chunk.size`, or `m_client->m_mesh_chunk.getMeshPos(...)` etc.

Not opposed, can surely do this. Just don't see much value in it.

Reply:
I see the value in not adding new methods to Client class, and the new class can be trivially copied around to avoid calls across threads. It would also be testable separately from the codebase.

Reply:
Wanna do a PR on my branch?

Just to be clear you mean something like:
```
struct MeshChunk {
   u16 size;

   s16 getMeshPos(s16) {...}
   bool checkMeshPos() ...
   size_t getPosIndex(s16, s16, s16) ...
   ...
};
```
?

Reply:
Correct. I'll make a PR